spacemesh/v2/tx.proto

```diff
@@ -0,0 +1,114 @@
+syntax = "proto3";
+
+import "google/rpc/status.proto";
+
+package spacemesh.v2;
+
+message Transaction {
+  oneof versioned {
+    TransactionV1 v1 = 1;
+  }
+}
+
+// An immutable Spacemesh transaction.
+// do not include mutable data such as tx state or result.
+message TransactionV1 {
```

Comment:
this needs to include a status field:

```
  enum Status {
    UNKNOWN = 0; // block hasn't been processed yet
    SUCCESS = 1; // tx was fully applied
    FAILURE = 2; // tx failed, but base gas was paid
    INEFFECTIVE = 3; // tx wasn't processed, no gas paid
  }
```

Reply:
actually, I think we can just include the `TransactionResult` here, which includes these things.

Reply:
I want to separate transaction from mutable data like tx state and result.
That's why I've created wrapper around it. https://github.com/spacemeshos/api/blob/api-v2/spacemesh/v2/tx.proto#L90